python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -174,6 +177,7 @@ def num_processing(self):
     def add_output(self, ref: RefBundle) -> None:
         """Move a bundle produced by the operator to its outqueue."""
         self.outqueue.append(ref)
+        self.invalidate_cached_outqueue_memory()
```

Comment:
Instead of invalidate the cache and re-calcuate each time, I think it'd be better to just use a int value to keep track of the memory size. Basically, we can adjust the value every time when bundles are added to or taken from the queue. 
The reason is because, if the queue keeps changing frequently, this invalidation won't help much.
Also, since the queues are shared between parent the child ops, I guess we can also write a class for the queue to make the code easier to maintain.